core/src/libraries/BalanceSheet.sol

```diff
@@ -92,7 +91,7 @@ library BalanceSheet {
             isSus = metric > _manipulationThreshold(_effectiveCollateralFactor(sigma));
 
             a = uint160((sqrtMeanPriceX96 * SoladyMath.sqrt(1e18 - sigma)) / 1e9);
-            b = SafeCastLib.safeCastTo160((sqrtMeanPriceX96 * SoladyMath.sqrt(1e18 + sigma)) / 1e9);
+            b = uint160(SoladyMath.min((sqrtMeanPriceX96 * SoladyMath.sqrt(1e18 + sigma)) / 1e9, type(uint160).max));
```

Comment:
Instead of reverting when `b` exceeds uint160, we now clamp to it. This ensures massive prices won't brick the contract.